src/System.Private.Xml/tests/XmlReader/Tests/ReaderEncodingTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Xunit;
+
+namespace System.Xml.Tests
+{
+    public class ReaderEncodingTests
+    {
+        [Fact]
+        public static void ReadWithSurrogateCharAndInvalidChar()
+        {
+            var bytes = new byte[] { 60, 0, 0, 0, 0, 34, 1, 0, 62, 100, 60, 47, 97, 62, 10 };
+            var reader = XmlReader.Create(new MemoryStream(bytes));
+
+            XmlException ex = Assert.Throws<XmlException>(() => reader.Read());
+            Assert.Equal("Invalid character in the given encoding. Line 1, position 4.", ex.Message);
+            Assert.Equal(4, ex.LinePosition);
+        }
+
+        [Fact]
+        public static void ReadWithNormalCharAndInvalidChar()
+        {
+            var bytes = new byte[] { 60, 0, 0, 0, 65, 0, 0, 0,62, 100, 60, 47, 97, 62, 10 };
+            var reader = XmlReader.Create(new MemoryStream(bytes));
+
+            XmlException ex = Assert.Throws<XmlException>(() => reader.Read());
+            Assert.Equal("Invalid character in the given encoding. Line 1, position 3.", ex.Message);
+            Assert.Equal(3, ex.LinePosition);
+        }
+
+        [Fact]
+        public static void ReadWithSurrogateCharAndInvalidChar_ValidXmlStructure()
+        {
+            var bytes = new byte[] { 60, 0, 0, 0, 97, 0, 0, 0, 62, 0, 0, 0, 0, 34, 1, 0, 62, 100, 60, 47, 60, 0, 0, 0, 47, 0, 0, 0, 97, 0, 0, 0, 62, 0, 0, 0 };
```

Comment:
- all of these tests start with `60, 0, 0, 0`  should you check for invalid characters further in the buffer as well?
- I'm missing test where you try to make invalid character on the threshold of the buffer size
- positive cases proving you didn't break existing correct surrogate chars scenarios

Reply:
> all of these tests start with 60, 0, 0, 0  should you check for invalid characters further in the buffer as well?

Didn't understood what you mean by "further in the buffer"

> I'm missing test where you try to make invalid character on the threshold of the buffer size

"on the threshold of the buffer size" is this mean at the end or beginning of byte array?

> positive cases proving you didn't break existing correct surrogate chars scenarios

There is one positive case https://github.com/buyaa-n/corefx/pull/1/files#diff-47e5e9b8251940e29f847159c2fd2fb4R55